operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -197,10 +201,28 @@ func (r *ReconcileIstioOperator) Reconcile(request reconcile.Request) (reconcile
 	var err error
 	iopMerged := *iop
 	iopMerged.Spec, err = helmreconciler.MergeIOPSWithProfile(iop.Spec)
+
 	if err != nil {
 		log.Errorf("failed to generate IstioOperator spec, %v", err)
 		return reconcile.Result{}, err
 	}
+
+	if _, ok := iopMerged.Spec.Values["global"]; !ok {
+		iopMerged.Spec.Values["global"] = make(map[string]interface{})
+	}
+	globalValues := iopMerged.Spec.Values["global"].(map[string]interface{})
+	log.Info("Detecting third-party JWT support")
+	var jwtPolicy util.JWTPolicy
+	if jwtPolicy, err = util.DetectSupportedJWTPolicy(r.config); err != nil {
```

Comment:
if we detect first party jwt then later there is a failure to connect to API server this may flip flop and break things. I have a pr out to improve the error handling but it may still happen in some weird case?

Reply:
That's true. We could write what we detected to .status and only run the detection once? I'm afraid that would need an API change, though...

Reply:
yeah I am not quite sure a good solution. In the 1.5 timeframe if we get my other pr in as well I am not too worried and we can improve on this in the future